src/components/shared/Modal.tsx

```diff
@@ -20,26 +20,56 @@ export type InputFields = {
   address: string;
 };
 
-export function Modal({ closeModal, addPatient }: ModalProps) {
+export function Modal({ closeModal, addPatient, currentPatient }: ModalProps) {
   const {
     register,
     handleSubmit,
+    setValue,
+    reset,
     formState: { errors },
   } = useForm<InputFields>({
     // resolver: yupResolver(RegistrationSchema),
   });
 
+  const { createPatient, editPatient } = usePatientsActions();
+
+  useEffect(() => {
+    if (currentPatient) {
+      const [firstName = "", lastName = ""] = currentPatient.name.split(" ");
+      setValue("firstName", firstName);
+      setValue("lastName", lastName);
+      setValue("idNumber", currentPatient.id || "");
+      setValue("dateBirth", currentPatient.dateBirth || "");
+      setValue("mobile", currentPatient.mobile || "");
+      setValue("email", currentPatient.email || "");
+      setValue("address", currentPatient.address || "");
+    } else {
+      reset();
+      reset({
+        firstName: "",
+        lastName: "",
+        idNumber: "",
+        dateBirth: "",
+        mobile: "",
+        email: "",
+        address: "",
+      });
+    }
+  }, [currentPatient, setValue, reset]);
```

Comment:
Zamiast robić to w ten sposób, prawdopodobnie możesz ustawić defaultValues w hooku useForm, będzie łatwiej. I nie musisz robić if elsa, po prostu dodawaj currentPatient?.id || "" i analogicznie całą resztę.